rules/utils/extend-fix-range.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+/**
+Extend fix range to prevent changes from other rules
+https://github.com/eslint/eslint/pull/13748/files#diff-c692f3fde09eda7c89f1802c908511a3fb59f5d207fe95eb009cb52e46a99e84R348
+*/
+function * extendFixRange(range = []) {
+	const [start = 0, end = Infinity] = range;
```

Comment:
I think the `range` should be mandatory since it's supposed to be within the source code.

`end = Infinity` works by chance now, but we could make a change that would break this.